### PR TITLE
set translatesAutoresizingMaskIntoConstraints when adding subview

### PR DESCRIPTION
**Pull request by overovermind**
Added a general helper for adding subviews when using constraints.

Before:
```
let subview = UIView()
subview.translatesAutoresizingMaskIntoConstraints = false
view.addSubview(subview)
```

Now:
```
let subview = UIView()
view.addSubview(subview, translatesAutoresizingMaskIntoConstraints: false)
```

Also possibile with array of views. Would love to see this added 😀

## Checklist

- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x ] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.
- [x] I have added a [changelog](https://github.com/SwifterSwift/SwifterSwift/blob/master/CHANGELOG_GUIDELINES.md) entry describing my changes.

### CODE REVIEW

File: Sources/Extensions/UIKit/UIViewExtensions.swift [hunk 1]
```diff
@@ -280,6 +280,23 @@ public extension UIView {
 	public func addSubviews(_ subviews: [UIView]) {
 		subviews.forEach({ self.addSubview($0) })
 	}
+    
+    /// SwifterSwift: Add array of subviews to view and set translatesAutoresizingMaskIntoConstraints.
+    ///
+    /// - Parameter subviews: array of subviews to add to self.
+    /// - Parameter translatesAutoresizingMaskIntoConstraints: set translatesAutoresizingMaskIntoConstraints of the added subviews.
+    public func addSubviews(_ subviews: [UIView], translatesAutoresizingMaskIntoConstraints: Bool) {
+        subviews.forEach({ self.addSubview($0, translatesAutoresizingMaskIntoConstraints: translatesAutoresizingMaskIntoConstraints) })
```
Comment (marcocapano) [inline Sources/Extensions/UIKit/UIViewExtensions.swift]: Nothing to say, it's simple as it should be. Maybe we should consider removing that "self" before addSubview ?
I'm not a fan of it when unnecessary and in the single view and in the methods it is not adopted. I feel like we should follow a common style.

Comment (overovermind) [reply Sources/Extensions/UIKit/UIViewExtensions.swift]: sure, done! Fixed it in the other `addSubviews` method as well

**Comment by overovermind** [comment]
When working with constraints in code, you always have to set `translatesAutoresizingMaskIntoConstraints` to `false` (the default is `true` for programmatically created views: https://developer.apple.com/documentation/uikit/uiview/1622572-translatesautoresizingmaskintoco). 

So you save some lines of code with this extension, as you don't have to set `translatesAutoresizingMaskIntoConstraints` explicitly for every view.

**Comment by guykogus** [comment]
I understand all that, but do you have examples in your code that you can show where you're really saving yourself a lot of code? I use plenty of programmatically-created views and I don't think I have to set that value a lot. My only worry is about bloating the code, but if you can show me an example I'll happily accept it.

**Comment by overovermind** [comment]
ok, here is an example:

```swift
override init(frame: CGRect) {
        super.init(frame: frame)
        
        addSubview(containerView, translatesAutoresizingMaskIntoConstraints: false)
        containerView.topAnchor.constraint(equalTo: layoutMarginsGuide.topAnchor, constant: 0).isActive = true
        containerView.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor, constant: 0).isActive = true
        containerView.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor, constant: 0).isActive = true
        containerView.heightAnchor.constraint(equalTo: containerView.widthAnchor, multiplier: 1, constant: 0).isActive = true
        
        addSubview(titleLabel, translatesAutoresizingMaskIntoConstraints: false)
        titleLabel.topAnchor.constraint(equalTo: containerView.bottomAnchor, constant: 0).isActive = true
        titleLabel.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor, constant: 0).isActive = true
        titleLabel.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor, constant: 0).isActive = true
        titleLabel.numberOfLines = 0
        titleLabel.lineBreakMode = .byWordWrapping
        titleLabel.textColor = .white
        titleLabel.textAlignment = .center
        titleLabel.font = .boldSystemFont(ofSize: 13)
        
        addSubview(captionLabel, translatesAutoresizingMaskIntoConstraints: false)
        captionLabel.topAnchor.constraint(equalTo: titleLabel.bottomAnchor, constant: 0).isActive = true
        captionLabel.leadingAnchor.constraint(equalTo: layoutMarginsGuide.leadingAnchor, constant: 0).isActive = true
        captionLabel.trailingAnchor.constraint(equalTo: layoutMarginsGuide.trailingAnchor, constant: 0).isActive = true
        captionLabel.bottomAnchor.constraint(equalTo: layoutMarginsGuide.bottomAnchor, constant: 0).isActive = true
        captionLabel.numberOfLines = 0
        captionLabel.lineBreakMode = .byWordWrapping
        captionLabel.textColor = .white
        captionLabel.textAlignment = .center
        captionLabel.font = .systemFont(ofSize: 12)
        
        containerView.addSubview(imageView, translatesAutoresizingMaskIntoConstraints: false)
        imageView.fillToSuperview()
        imageView.contentMode = .scaleAspectFill
        imageView.clipsToBounds = true
        
        containerView.addSubview(button, translatesAutoresizingMaskIntoConstraints: false)
        button.fillToSuperview()
    }
```

Feel free to decide if this should be added 😃

**Comment by guykogus** [comment]
Oh wow what is that?! 🤣 Why isn't that coming from a XIB?
I mean yeah, you have 38 lines instead of 43, so makes sense to save yourself that little bit.

File: Tests/UIKitTests/UIViewExtensionsTests.swift [hunk 2]
```diff
@@ -206,6 +206,38 @@ final class UIViewExtensionsTests: XCTestCase {
 		view.addSubviews([UIView(), UIView()])
 		XCTAssertEqual(view.subviews.count, 2)
 	}
+    
+    func testAddSubViewsWithTranslatesAutoresizingMaskIntoConstraints() {
```
Comment (guykogus) [inline Tests/UIKitTests/UIViewExtensionsTests.swift]: Can you please use multiple values in the array just to make it a bit clearer that it works on multiple sizes. Also test for an empty array.

File: Tests/UIKitTests/UIViewExtensionsTests.swift [hunk 3]
```diff
@@ -206,6 +206,38 @@ final class UIViewExtensionsTests: XCTestCase {
...
+        let frame = CGRect(x: 0, y: 0, width: 100, height: 100)
+        let view = UIView(frame: frame)
```
Comment (guykogus) [inline Tests/UIKitTests/UIViewExtensionsTests.swift]: The frame's irrelevant for these tests, so you can just use `UIView()` like you do later down.
This comment applies to `testAddSubViewWithTranslatesAutoresizingMaskIntoConstraints` as well.

**Comment by LucianoPAlmeida** [comment]
Hey @overovermind Thank's for taking the time for this PR 👍 
But to me to have a method that will modify the state of an instance that is passed as parameter before adding as subview seems a little weird. My suggestion for that will be instead of a addSubview() create a convenience initializer for the view that takes this parameter, to construct it with this behavior, like your PR description example the now will be like
```swift

let subview = UIView(translatesAutoresizingMaskIntoConstraints: false)
view.addSubview(subview)
```
I believe this provides the same convenience, but makes the propurse more clear :)
Let me know what your thougths on that :))

**Comment by guykogus** [comment]
@LucianoPAlmeida I agree, that seems like a better way to do it. The problem, I believe, is that both views created programmatically and from XIBs (root view only) start with the value set to `true`, so you'd only be solving for the first instance.

I'm still not sure about this extension at the end of the day. Can we maybe get some more opinions?

**Comment by LucianoPAlmeida** [comment]
@guykogus @overovermind  I see the point there, just trying to find better options for these extension since I'm not  sure this extension fits to SwifterSwift :)) 
cc @SD10 @omaralbeik pinging for thoughts on that

**Comment by SD10** [comment]
Hey @overovermind, thanks for taking the time to submit this extension 💯 👍 

Unfortunately, I don't think this is a good fit as an extension for SwifterSwift because adding a subview and setting `translatesAutoresizingMaskIntoConstraints: false` to false are unrelated in my opinion.

Maybe you could open up another PR with a similar extension:
```swift
extension NSLayoutConstraint {
    static func prepareViews(_ views: [UIView]) {
        views.forEach { $0.translatesAutoresizingMaskIntoConstraints = false }
    }
}
```

**Comment by kooraz** [comment]
guykogus, do you actively work as an iOS Developer? If so, you must be a Trumpist aswell. Some clicky clicky and fuck all other Developers trying to refactor my code. Grow up, this is an awesome feature!

**Comment by kooraz** [comment]
If you actually use XIBs in prod, go start react native, feels like the perfect setting for you.

**Comment by omaralbeik** [comment]
@kooraz, Althoght I don't like interface builder, I disagree with you here.
Lyft is an example for a very successful app that utlilizes Interface builder, both options are used in the industry.

Second, please don't be disrespectful to other contributors, such a language is not accepted here. please refer to our [code of conduct](https://github.com/SwifterSwift/SwifterSwift/blob/master/.github/CODE_OF_CONDUCT.md) for further details and try to be nicer to others!

**Comment by guykogus** [comment]
@kooraz I've been actively working as an iOS developer for 9 years.
I almost exclusively work with XIBs and storyboards. I'd be curious to know why you're so against them.
I've used react native in prod code, but it's very limiting (and frustrating!) so it was only for a small part of the app. Airbnb wrote a [detailed article](https://medium.com/airbnb-engineering/react-native-at-airbnb-f95aa460be1c) about their experience with it explaining the pros and cons and why they eventually discontinued using it. You should give it a read.

I know it must be frustrating to have your PR closed, but what may be a great extension for you, with your way of working, may not be for others. We have all discussed and agreed that this was not a good fit for SwifterSwift. I had a similar feeling with one of my first PRs for this repo that got rejected (and also behaved inappropriately, to be honest), and it's not the right way to go about it. I realised that this repo isn't for every little additional function or utility that we use in our apps, it's meant much more for functions that you look at and think, "this should be part of the Swift standard library". If you want a recent example look at [this PR for CGVector](https://github.com/SwifterSwift/SwifterSwift/pull/527).

**Comment by overovermind** [comment]
Thanks for all the feedback 👍 I will think of another solution and open a new PR
